src/main/java/seedu/recipe/logic/commands/FavouriteCommand.java

```diff
@@ -36,31 +39,66 @@ public FavouriteCommand(Index[] targetIndex) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Recipe> lastShownList = model.getFilteredRecipeList();
-        StringBuilder sb = new StringBuilder().append("Added ");
+        List<String> successfulFavouritesList = new ArrayList<>();
+        List<String> alreadyFavouritesList = new ArrayList<>();
 
-        for (int i = 0; i < targetIndex.length; i++) {
-            if (targetIndex[i].getZeroBased() >= lastShownList.size()) {
-                throw new CommandException(Messages.MESSAGE_INVALID_RECIPE_DISPLAYED_INDEX);
-            }
+        if (!canFavouriteTargetRecipes(lastShownList.size(), targetIndex)) {
+            throw new CommandException(Messages.MESSAGE_INVALID_RECIPE_DISPLAYED_INDEX);
+        }
 
-            Recipe recipeToFavourite = lastShownList.get(targetIndex[i].getZeroBased());
-            model.favouriteRecipe(recipeToFavourite);
-            if (i == targetIndex.length - 1 && targetIndex.length != 1) {
-                sb.append(" and ");
-            }
-            sb.append(recipeToFavourite.getName().toString());
-            if (i < targetIndex.length - 2) {
-                sb.append(", ");
+        for (Index index : targetIndex) {
+            Recipe recipeToFavourite = lastShownList.get(index.getZeroBased());
+            if (!recipeToFavourite.isFavourite()) {
+                model.favouriteRecipe(recipeToFavourite);
```

Comment:
Hey Brian, I was wondering whether we could format the favourite command (and the unfavourite command) to be similar to the edit command, where it searches for the Recipe to favourite, creates a new Recipe object with favourite = true, and calls model.setRecipe instead.
Currently, the model.favouriteRecipe method would create a new Recipe in the UniqueRecipeList class, and im thinking that it might not be good to create a new Recipe in this class.

Reply:
Thanks for pointing this out! I did this ages ago before I could appreciate the usage of EditRecipeDescriptor. Fixed in latest PR.